woocommerce/woocommerce
# fix: wcadmin react18 createroot marketing coupons

Related to #46879

Additional information on this issue can be found in [How to Upgrade to React 18](https://react.dev/blog/2022/03/08/react-18-upgrade-guide) and in [Upgrading to React 18 and common pitfalls of concurrent mode](https://make.wordpress.org/core/2023/03/07/upgrading-to-react-18-and-common-pitfalls-of-concurrent-mode/).

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/50528.

In this PR, we replace `ReactDOM.render` with `createRoot`, as per [React 18 upgrade guide](https://react.dev/blog/2022/03/08/react-18-upgrade-guide). This fixes the "ReactDOM.render is no longer supported in React 18. Use createRoot instead." error message in browser console.

Screenshot: 

<img width="1724" alt="pr-no-react-console-error" src="https://github.com/user-attachments/assets/64613db4-7efb-47d2-a1ce-d0324f941fc1">



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open browser console.
2. Go to Marketing > Coupons page: `/wp-admin/edit.php?post_type=shop_coupon`
3. There should be no console error that says "ReactDOM.render is no longer supported in React 18. Use createRoot instead."



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
